lib/puma/request.rb

```diff
@@ -464,8 +464,15 @@ def str_headers(env, status, headers, res_info, lines, requests, client)
           next
         end
 
-        if vs.respond_to?(:to_s) && !vs.to_s.empty?
-          vs.to_s.split(NEWLINE).each do |v|
+        ary = if vs.is_a?(::Array) && !vs.empty?
+          vs
+        elsif vs.respond_to?(:to_s) && !vs.to_s.empty?
+          vs.to_s.split NEWLINE
+        else
+          nil
```

Comment:
if we just set this to `[]` then we can skip the `if ary` check

Reply:
Thanks.  Done.

Reply:
This broke CI, I'm going to revert.